master/internal/hpimportance/worker.go

```diff
@@ -99,7 +117,16 @@ func taskHandlerFactory(db *db.PgDB, system *actor.System) func(interface{}) int
 			sendWorkFailed(system, work, "invalid metric type received in hyperparameter importance worker")
 			return nil
 		}
-		results := computeHPImportance(trials)
+		taskDir := path.Join(workingDir, fmt.Sprint(actorId))
+		err = os.Mkdir(taskDir, 0066)
+		if err != nil {
+			sendWorkFailed(system, work, err.Error())
+		}
+		results := computeHPImportance(trials, experimentConfig, masterConfig, growforest, taskDir)
+		err = os.RemoveAll(taskDir)
+		if err != nil {
+			ctx.Log().Errorf("Failed to clean up temporary directory %s", taskDir)
```

Comment:
dont forget a with err so we dont lose what actually happened
```suggestion
			ctx.Log().WithErr(err).Errorf("Failed to clean up temporary directory %s", taskDir)
```